src/main/java/de/numcodex/feasibility_gui_backend/query/result/ResultService.java

```diff
@@ -0,0 +1,77 @@
+package de.numcodex.feasibility_gui_backend.query.result;
+
+import com.github.benmanes.caffeine.cache.Cache;
+import de.numcodex.feasibility_gui_backend.query.persistence.ResultType;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map.Entry;
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+import lombok.Synchronized;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+@Service
+@Slf4j
+@RequiredArgsConstructor
+public class ResultService {
+
+  @NonNull
+  private Cache<Long, QueryResult> queryResultCache;
+
+  public QueryResult getResult(Long queryId) {
+    return queryResultCache.getIfPresent(queryId);
+  }
+
+  @Synchronized
+  public void addResultLine(Long queryId, ResultLine resultLine) {
```

Comment:
You can use `cache.asMap().merge(...)` instead of doing your own synchronisation. Also `QueryResult` should be immutable in order to avoid changing a `QueryResult` already returned by this service.

Reply:
I did it to my best knowledge...but I am really not sure if this is correct. It looks a bit cumbersome...

Reply:
I added a commit with the merge implemented in QueryResult itself.